web/src/user_topics_ui.js

```diff
@@ -13,11 +12,18 @@ import * as user_topics from "./user_topics";
 export function handle_topic_updates(user_topic_event) {
     // Update the UI after changes in topic visibility policies.
     user_topics.set_user_topic(user_topic_event);
-    popover_menus.get_topic_menu_popover()?.hide();
 
-    stream_list.update_streams_sidebar();
-    unread_ui.update_unread_counts();
-    message_lists.current?.update_muting_and_rerender();
+    // These calls update the reference elements of the popovers containing
+    // the topic visibility policy switcher, which can lead to the force
+    // closing of these popovers. We add a delay, the same as that of the
+    // topics visibility popover switcher, to allow animations to show through
+    // without being interrupted.
+    setTimeout(() => {
+        stream_list.update_streams_sidebar();
+        unread_ui.update_unread_counts();
+        message_lists.current?.update_muting_and_rerender();
+    }, 500);
```

Comment:
This delay seems like it could be conditional on the popover being open? I don't see a reason to defer it in other cases.

Reply:
The reason for adding this delay here is that these methods are responsible for hiding/updating the reference element of the popover when a topic is muted. This leads to the force closing of the popovers since their reference element is no longer in the view. Thus, we are left with half-fulfilled animations.

For more clarity, I have separated this into its own commit (d60fb05416c4f8ae2935d84625281f4fab6f1468), added a better comment on the "why" and restricted this delay to only come into action when a topic is muted.

Reply:
I have also included `recent_view_ui.update_topic_visibility_policy` into this delay group since that also removed the topic from view when muted. I missed this the last time.